File: server/data/results/translations/lite/plain/cpp/bubble_sort.cpp
#include <iostream>
#include <vector>
#include <string>

std::vector<int> bubble_sort(const std::vector<int>& values) {
    std::vector<int> items = values;
    int n = items.size();
    for (int i = 0; i < n; ++i) {
        bool swapped = false;
        for (int j = 0; j < n - i - 1; ++j) {
            if (items[j] > items[j + 1]) {
                int temp = items[j];
                items[j] = items[j + 1];
                items[j + 1] = temp;
                swapped = true;
            }
        }
        if (!swapped) {
            break;
        }
    }
    return items;
}

void main_func() {
    std::vector<int> data = {5, 2, 9, 1, 5, 6};
    std::vector<int> result = bubble_sort(data);
    std::string line = "";
    for (int value : result) {
        line = line + std::to_string(value) + " ";
    }
    
    // strip() equivalent: trim trailing space if not empty
    if (!line.empty() && line.back() == ' ') {
        line.pop_back();
    }
    
    std::cout << line << std::endl;
}

int main() {
    main_func();
    return 0;
}

File: server/data/results/translations/lite/plain/java/bubble_sort.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Writing {

    public static List<Integer> bubble_sort(List<Integer> values) {
        List<Integer> items = new ArrayList<>(values);
        int n = items.size();
        for (int i = 0; i < n; i++) {
            boolean swapped = false;
            for (int j = 0; j < n - i - 1; j++) {
                if (items.get(j).compareTo(items.get(j + 1)) > 0) {
                    int temp = items.get(j);
                    items.set(j, items.get(j + 1));
                    items.set(j + 1, temp);
                    swapped = true;
                }
            }
            if (!swapped) {
                break;
            }
        }
        return items;
    }

    public static void main(String[] args) {
        List<Integer> data = Arrays.asList(5, 2, 9, 1, 5, 6);
        List<Integer> result = bubble_sort(data);
        StringBuilder line = new StringBuilder();
        for (Integer value : result) {
            line.append(value).append(" ");
        }
        System.out.println(line.toString().trim());
    }
}